plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -228,17 +275,28 @@ synchronized void updatePipelines() throws Exception {
         }
     }
 
-    void startUpdateWorker() {
-        threadPool.schedule(pipelineUpdateInterval, ThreadPool.Names.GENERIC, new Updater());
-    }
-
     boolean existPipeline(String pipelineId) {
         GetRequest request = new GetRequest(PipelineStore.INDEX, PipelineStore.TYPE, pipelineId);
-        GetResponse response = client().get(request).actionGet();
-        return response.isExists();
+        try {
+            GetResponse response = client().get(request).actionGet();
+            return response.isExists();
+        } catch (IndexNotFoundException e) {
+            // the ingest index doesn't exist, so the pipeline doesn't either:
+            return false;
+        }
     }
 
-    Iterable<SearchHit> readAllPipelines() {
+    void initialize() {
+        try {
+            innerUpdatePipelines();
+        } catch (Exception e) {
+            logger.error("Failed to load the pipelines during startup");
+        }
+        pipelineIndexReady = true;
+        logger.debug("Pipeline store started with [{}] pipelines", pipelines.size());
+    }
+
+    private Iterable<SearchHit> readAllPipelines() {
         // TODO: the search should be replaced with an ingest API when it is available
```

Comment:
Yes, that makes sense. I think this api should be named list pipeline API? Which under the hood just executes a scroll search. Although I think it will be phase2?